### PR TITLE
fix(chat): elapsed timer resets when switching between workspaces

### LINKED ISSUE #359: fix(chat): elapsed timer resets when switching between workspaces
## Summary

The elapsed time indicator shown at the bottom of the chat during an active agent prompt resets to zero when you switch away from the workspace and switch back. It should reflect the true wall-clock time since the prompt started, regardless of whether the user stays on that workspace or navigates between them.

## Steps to Reproduce

1. Start a prompt in Workspace A — elapsed timer begins counting up
2. Switch to Workspace B
3. Switch back to Workspace A
4. The elapsed timer has reset to 0:00 instead of showing the actual elapsed time

## Expected Behavior

The timer should show the true elapsed time since the prompt started. Switching workspaces and coming back should display the correct accumulated time (e.g., if 45 seconds passed, it should show 0:45 not 0:00).

## Actual Behavior

The timer resets on workspace switch, likely because it's driven by a local component `useState` / `useEffect` interval that reinitializes when the component remounts.

## Suggested Approach

Store the prompt start timestamp (e.g., `Date.now()`) in the Zustand store keyed by workspace ID rather than relying on a component-local timer. The display component computes elapsed time as `now - startTimestamp` on each tick, so remounting the component naturally shows the correct value.

### PR DESCRIPTION
**Pull request by doomspork**
## Summary

The elapsed time indicator at the bottom of the chat resets to 0:00 when switching workspaces and switching back. The root cause is a component-local `useRef` in `ChatPanel` that captures a fresh `Date.now()` on every remount.

This PR moves the prompt start timestamp into the Zustand store keyed by workspace ID. The timer now computes elapsed as `Date.now() - promptStartTime` on each tick, so switching away and back shows the correct wall-clock time.

- **`useAppStore.ts`** — Added `promptStartTime: Record<string, number>` with set/clear actions
- **`ChatPanel.tsx`** — Replaced `useRef`-based timer with store-backed `promptStartTime`; set on prompt send, clear on send failure
- **`useAgentStream.ts`** — Clear `promptStartTime` on `ProcessExited` and `result` events

## Complexity Notes

The timestamp is intentionally **not** reset during compaction (Running → Compacting → Running) since `isAgentBusy()` returns true for both states — the timer continues seamlessly across compaction pauses.

## Test Steps

1. Start a prompt in Workspace A — verify the elapsed timer counts up
2. Switch to Workspace B, wait ~15 seconds
3. Switch back to Workspace A — verify the timer shows ~15s (not 0:00)
4. Wait for the prompt to finish — verify the timer disappears normally
5. Start another prompt — verify the timer starts fresh from 0:00
6. (Edge case) If a prompt errors on send, verify no stale timer remains

## Checklist

- [x] Tests pass (608/608 vitest, tsc --noEmit clean)
- [ ] Manual testing with workspace switching

Closes #359

### CODE REVIEW

File: src/ui/src/components/chat/ChatPanel.tsx [hunk 1]
```diff
@@ -291,27 +291,21 @@ export function ChatPanel() {
...
-    if (!isRunning) {
-      startTimeRef.current = null;
-      return;
-    }
-    if (!startTimeRef.current) {
-      startTimeRef.current = Date.now();
-      setElapsed(0);
-      setSpinnerIdx(0);
-    }
+    if (!isRunning || !promptStartTime) return;
```
[the remark this answers is not in the document]
Comment (doomspork) [reply src/ui/src/components/chat/ChatPanel.tsx]: Done — switched to `promptStartTime == null` so valid numeric timestamps (including 0) aren't accidentally treated as missing.

File: src/ui/src/stores/useAppStore.ts [hunk 2]
```diff
@@ -593,6 +596,18 @@ export const useAppStore = create<AppState>((set) => ({
...
+    set((s) => ({
+      promptStartTime: { ...s.promptStartTime, [wsId]: time },
+    })),
+  clearPromptStartTime: (wsId) =>
+    set((s) => {
+      if (!(wsId in s.promptStartTime)) return {};
+      const next = { ...s.promptStartTime };
+      delete next[wsId];
+      return { promptStartTime: next };
+    }),
```
[the remark this answers is not in the document]
Comment (doomspork) [reply src/ui/src/stores/useAppStore.ts:610]: Done — added tests for `setPromptStartTime`, `clearPromptStartTime`, per-workspace isolation, overwrite behavior, and no-op clear.
